lightning/src/ln/msgs.rs

```diff
@@ -1698,6 +1698,13 @@ mod fuzzy_internal_msgs {
 			amt_to_forward: u64,
 			outgoing_cltv_value: u32,
 		},
+		#[allow(unused)]
+		TrampolineEntry {
+			amt_to_forward: u64,
+			outgoing_cltv_value: u32,
+			payment_data: Option<FinalOnionHopData>,
```

Comment:
All of those fields work exactly the same as for normal payments, as the behavior of an intermediate trampoline node, 
when processing HTLCs that contain a trampoline onion is to:

- first act as if it were a payment recipient, aggregating incoming payment parts
- then forward the "received" payment to a relayer component instead of treating it to a payment to us

Rusty also had a question about the `payment_secret` in the outer onion, as that part is a bit subtle: https://github.com/lightning/bolts/pull/836#discussion_r566557686.

In eclair we use the following trait hierarchy: https://github.com/ACINQ/eclair/blob/c8184b3e43a603454d672aed61b2ae461a80e1a8/eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/PaymentOnion.scala#L197. Once decrypted, we combine the outer and inner onion information received by an intermediate trampoline node as `NodeRelayPayload`, to show that this data is used for relay.

> @t-bast, do you figure it'd be worthwhile expanding a bit in the https://github.com/lightning/bolts/pull/836 on how the roles of the fields in the outer onion are altered in a Trampoline context?

I thought that was already detailed well enough in the requirements section of the specification, if there is a specific point you'd like to be more detailed can you put a comment on the spec?

> As well as any arithmetic considerations for routing fees for a path whose length isn't known a priori?

That's something I described in the "trampoline fees" section of this spec PR: https://github.com/lightning/bolts/pull/829
Let me know directly on that PR if the section is unclear or if you disagree with some of it. In eclair we previously used a very basic trial-and-error approach (start with a low fee, increase it on errors), and recently moved to a different model where the sender always uses 4% for the trampoline fees, and the trampoline node tries to deal with that (and could in the future do it at a loss in some payments as long as other payments compensate that loss).